core/src/main/scala/quasar/destination/avalanche/UserInfoGetter.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.avalanche
+
+import argonaut._
+
+import cats._
+import cats.effect._
+import cats.implicits._
+
+import java.lang.String
+import java.nio.charset.StandardCharsets
+import java.util.UUID
+
+import org.http4s.{Request, Uri, Method, Headers, AuthScheme}
+import org.http4s.headers.Authorization
+import org.http4s.ember.client.EmberClientBuilder
+import org.http4s.argonaut.jsonDecoder
+import org.http4s.syntax.literals._
+
+import scala.{Option, Some, None, StringContext, Either, Left, Right}
+
+import quasar.connector.{Credentials, GetAuth, ExternalCredentials}
+
+object UserInfoGetter {
+
+  private val utf8 = StandardCharsets.UTF_8
+
+  private def emailFromUserinfo[F[_]: ConcurrentEffect: Timer: ContextShift](token: Credentials.Token, userinfoUrl: Uri): F[Option[Email]] = {
+    val req = Request[F](
+      uri = userinfoUrl,
+      method = Method.GET,
+      headers = Headers.of(
+        Authorization(
+          org.http4s.Credentials.Token(AuthScheme.Bearer, new String(token.toByteArray, utf8))
+        )
+      )
+    )
+
+    EmberClientBuilder
+      .default[F]
+      .build
```

Comment:
> Maybe separate the build part from the usage part? (So that the client is reused across multiple requests rather than recreated every request)

The idea at the moment is that the request is only made once at initialisation stage and never required again, hence the one-and-done implementation. This may change depending on how testing goes though, but I don't know until I _can_ test.

Reply:
I'm usually trying to define/create things "in the right place" so that the setup becomes less dependent on changes in circumstances (like the number of requests here). In particular, what I probably would have done here is create a class with the client as a parameter in the constructor.

But yeah, that's just my personal preference, totally fine if you prefer to leave it like this.

Reply:
I'll leave it like this for now and revisit the topic after I've done some testing